### DISCUSSION TITLE
Is it possible to terminate a request without proxying it within the proxy service?

### DISCUSSION THREAD

**Discussion by gduffy8** [dotnet/yarp#818] [General] [answered]
Hi,

I know it is simple to add an endpoint to our service where we dont have to leverage the YARP middleware. However, I am curious, can we use the YARP configuration to simply terminate a request without proxying it to another service, and return a 200?

A use case I am thinking is to be able to dynamically update the config with such an option, without having to explicitly add a new endpoint in code. So we could leverage whatever middleware has been loaded, without having to forward the request anywhere. I hope that makes sense.

Thanks

**Comment by Tratcher** [comment, marked as answer]
That makes sense but is not currently supported by default. Here are a few ways you can go about it:

- Have your own config section that you read to generate terminal routes.
- Use the YARP config and add a [middleware](https://github.com/microsoft/reverse-proxy/blob/ac296206f66b52ee182d336ac3129bb2a889a952/samples/ReverseProxy.Code.Sample/Startup.cs#L50) in the yarp pipeline that checks for specific routes and terminates them (e.g. doesn't call next).

**Reply by samsp-msft** [reply]
Option (b) with a custom tag in the route metadata would probably work well. Tratcher is doing (may have completed) some refactoring that would attach the route to the proxy feature, so this should be pretty simple. See the reverseproxy.code.sample for something pretty close.

**Reply by Tratcher** [reply]
https://github.com/microsoft/reverse-proxy/pull/814

**Reply by gduffy8** [reply]
Nice guys. Yes we were going down the path of adding a custom flag within the metadata of the route and writing some middleware that checks for it. Allows us to configure this route on the fly (if necessary). Thanks, I'll check out #814
